tests/v2/actions/harvester/harvester.go

```diff
@@ -0,0 +1,133 @@
+package harvester
+
+import (
+	"time"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/rancher/norman/types"
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+
+	"github.com/rancher/shepherd/clients/harvester"
+	"github.com/rancher/shepherd/clients/rancher"
+
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	Harvester            = "harvester"
+	HarvesterSettingType = "harvesterhci.io.setting"
+
+	clusterRegistrationURL = "cluster-registration-url"
+	clusterId              = "clusterId"
+	providerLabel          = "provider.cattle.io"
+)
+
+// RegisterHarvesterCluster is a function that creates a Virtualization Management object in rancher, and registers
+// the external harvester cluster with said object in rancher. This is required to use harvester as a rancher provider.
+func RegisterHarvesterWithRancher(rancherClient *rancher.Client, harvesterClient *harvester.Client) (string, error) {
+	importCluster := &provv1.Cluster{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      namegenerator.AppendRandomString("hvst-import"),
+			Namespace: "fleet-default",
+			Labels: map[string]string{
+				providerLabel: Harvester,
+			},
+		},
+	}
+
+	backoff := wait.Backoff{
+		Duration: 5 * time.Second,
+		Factor:   1.1,
+		Jitter:   0.1,
+		Steps:    40,
```

Comment:
since the same backoff is being used to get the cluster status, would steps:40 be sufficient here? as the cluster might take some time to come active after the registration command is run.

Reply:
I haven't seen any flakiness with this step during testing.